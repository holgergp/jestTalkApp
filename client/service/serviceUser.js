import { getPosts } from "../service/httpService";

export let doSomethingWithRemoteData = somePar => {
  const postData = getPosts();
  return postData.posts.map(it => it.toUpperCase());
};

export const uncoveredFunction = () => {
  const postData = getPosts();
  return postData.posts.map(it => it.toLowerCase());
}