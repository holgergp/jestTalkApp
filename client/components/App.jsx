import React from "react";
import List from "./ListComponent.jsx";
import { doSomethingWithRemoteData } from "../service/serviceUser.js";

const App = class App extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      posts: []
    };
  }

  render() {
    const { posts } = this.state;
    return (
      <div style={{ textAlign: "center" }}>
        <h1>Hello World</h1>
        <List values={posts} />
      </div>
    );
  }

  componentDidMount() {
    const posts = doSomethingWithRemoteData("param") || [];
    this.setState({ posts });
  }
};
export default App;
