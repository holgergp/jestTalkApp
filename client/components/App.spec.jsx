import App from "./App.jsx";
import React from "react";
import renderer from "react-test-renderer";
import * as dependency from "../service/serviceUser.js";

describe("App", () => {
  it("renders", () => {
    const renderedApp = renderer.create(<App />).toJSON();
    expect(renderedApp).toMatchSnapshot();
  });

  it("calling Service correctly", () => {
    dependency.doSomethingWithRemoteData = jest.fn();
    const renderedList = renderer.create(<App />).toJSON();

    expect(dependency.doSomethingWithRemoteData).toHaveBeenCalled();
    expect(dependency.doSomethingWithRemoteData).toBeCalled();

    expect(dependency.doSomethingWithRemoteData.mock.calls[0][0]).toBe("param");
    expect(dependency.doSomethingWithRemoteData).toBeCalledWith("param");
  });

  it("renders with defined data", () => {
    dependency.doSomethingWithRemoteData = jest
      .fn()
      .mockReturnValueOnce(["test"]);

    const renderedApp = renderer.create(<App />).toJSON();
    expect(renderedApp).toMatchSnapshot();
  });

  it("renders with defined return function", () => {
    dependency.doSomethingWithRemoteData = jest
      .fn()
      .mockImplementation(() => ["mockImplementation"]);

    const renderedApp = renderer.create(<App />).toJSON();
    expect(renderedApp).toMatchSnapshot();
  });
});
