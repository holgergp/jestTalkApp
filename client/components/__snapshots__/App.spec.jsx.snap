// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders 1`] = `
<div
  style={
    Object {
      "textAlign": "center",
    }
  }
>
  <h1>
    Hello World
  </h1>
  <dl>
    <dt>
      <div
        className="Text"
      >
        COMPLICATED
      </div>
    </dt>
    <dt>
      <div
        className="Text"
      >
        JSON
      </div>
    </dt>
    <dt>
      <div
        className="Text"
      >
        RESPONSE
      </div>
    </dt>
  </dl>
</div>
`;

exports[`App renders with defined data 1`] = `
<div
  style={
    Object {
      "textAlign": "center",
    }
  }
>
  <h1>
    Hello World
  </h1>
  <dl>
    <dt>
      <div
        className="Text"
      >
        test
      </div>
    </dt>
  </dl>
</div>
`;

exports[`App renders with defined return function 1`] = `
<div
  style={
    Object {
      "textAlign": "center",
    }
  }
>
  <h1>
    Hello World
  </h1>
  <dl>
    <dt>
      <div
        className="Text"
      >
        mockImplementation
      </div>
    </dt>
  </dl>
</div>
`;
